cmake/hipsycl-config.cmake.in

```diff
@@ -21,18 +148,8 @@ function(add_sycl_to_target)
     ${ARGN}
   )
 
-  if(HIPSYCL_PLATFORM_CANONICAL STREQUAL "cpu")
-    target_link_libraries(${ADD_SYCL_TARGET} PUBLIC hipSYCL::hipSYCL_cpu)
-  elseif(HIPSYCL_PLATFORM_CANONICAL MATCHES "cuda")
-    target_link_libraries(${ADD_SYCL_TARGET} PUBLIC hipSYCL::hipSYCL_cuda)
-  else()
-    target_link_libraries(${ADD_SYCL_TARGET} PUBLIC hipSYCL::hipSYCL_rocm)
-  endif()
+  set_target_properties("${ADD_SYCL_TARGET}" PROPERTIES RULE_LAUNCH_COMPILE "${HIPSYCL_SYCLCC_LAUNCH_RULE}")
+  set_target_properties("${ADD_SYCL_TARGET}" PROPERTIES RULE_LAUNCH_LINK "${HIPSYCL_SYCLCC_LAUNCH_RULE}")
 
-  # For now we have to link using syclcc, as CMake doesn't know about all
-  # the required libraries that need to be linked with the target.
-  set_target_properties(${ADD_SYCL_TARGET} PROPERTIES LINKER_LANGUAGE SYCL)
-
-  set_source_files_properties(${ADD_SYCL_SOURCES} PROPERTIES LANGUAGE SYCL)
+  target_link_libraries(${ADD_SYCL_TARGET} PUBLIC hipSYCL::hipSYCL-rt)
```

Comment:
Could you please elaborate why we need that? See also the discussion in PR #305. `syclcc` will already automatically link with the runtime, so why can we not just use it as linker and let it do its thing?

Reply:
One reason I can think of why this is useful is dependency tracking: whenever hipSYCL-rt changes, CMake projects that depend on it will recompile. If we don't have this line, CMake and the generated build system have no way of knowing that this dependency exists!

Reply:
It's also needed for CMake to know about the hipSYCL include directories. Without this, the include directories do not appear in the target's `CXX_FLAGS`, and clangd/IDEs cannnot resolve them.

Reply:
hmm.. Okay. I guess now that we only have to link against a single hipSYCL library, this has become much simpler and now there's very little duplication of logic by also linking from cmake. And it seems there are some benefits in doing it this way.